File: LucyBell.Server/Entidades/Pedido.cs
using Microsoft.AspNetCore.Identity;

namespace LucyBell.Server.Entidades
{
	public class Pedido
	{
		public int Id { get; set; }
		public string Estado { get; set; }
		public decimal Total { get; set; }
		public string MetodoPago { get; set; }
		public DateTime FechaCreacion { get; set; }
		public DateTime FechaActualizacion { get; set; }
		public bool EsEnvio { get; set; }
		public string UsuarioId { get; set; }
		public IdentityUser Usuario { get; set; }
		public List<DetallePedido> DetallesPedido { get; set; }
		public Envio? Envio { get; set; }
		public Retiro? Retiro { get; set; }
	}
}

File: lucybell.client/src/app/Models/Pedido.ts
export interface DetallePedidoDTO {
  cantidad: number;
  precioUnitario: number;
  productoId: number;
  varianteProductoId: number | null;
}

export interface EnvioDTO {
  direccion: string;
  barrio: string;
  codigoPostal: string;
  observacion: string;
  fechaEstimada: string;
}

export interface RetiroDTO {
  puntoRetiro: string;
  nombreRetira: string;
  documentoRetira: string;
  fechaRetiro: string;
}

export interface PedidoDTO {
  id: number;
  estado: string;
  total: number;
  metodoPago: string;
  fechaCreacion: string;
  fechaActualizacion: string;
  esEnvio: boolean;
  detalles: DetallePedidoDTO[];
  envio?: EnvioDTO | null;
  retiro?: RetiroDTO | null;
}

export interface PedidoCreacionDTO {
  estado: string;
  total: number;
  metodoPago: string;
  fechaCreacion: string;
  fechaActualizacion: string;
  esEnvio: boolean;
  usuarioId: string;
  detalles: DetallePedidoDTO[];
  envio?: EnvioDTO | null;
  retiro?: RetiroDTO | null;
}